docs/src/content/docs/book/functions.mdx

```diff
@@ -193,10 +472,544 @@ contract ManualMethodId {
+
+#### Static extension functions {#attributes-extends-static}
+
+There is no such attribute yet, so the special static extension functions in the standard library of Tact are introduced directly by the compiler:
+
+* [`Message.opcode(){:tact}`](/ref/core-cells#messageopcode)
+* [`Message.fromCell(){:tact}`](/ref/core-cells#messagefromcell)
+* [`Message.fromSlice(){:tact}`](/ref/core-cells#messagefromslice)
+* [`Struct.fromCell(){:tact}`](/ref/core-cells#structfromcell)
+* [`Struct.fromSlice(){:tact}`](/ref/core-cells#structfromslice)
+
+### Inheritance
+
+<Badge text="Contract scope" title="Functions can only be inherited from traits and never at the top level" variant="note" size="medium"/><p/>
+
+The `with{:tact}` keyword allows a contract to inherit a [trait](/book/types#traits) with all its constants, fields, and functions, including those transitively inherited from other traits associated with the specified trait.
+
+In particular, all functions of the trait become accessible in the contract, regardless of any inheritance-related attributes, such as `abstract{:tact}`, `virtual{:tact}`, or `override{:tact}`.
+
+You can allow a contract inheriting a [trait][trait] to modify a [internal function](#fun-internal) marked with the `virtual{:tact}` keyword by using `override{:tact}`. A function can also be marked as `abstract{:tact}`, in which case the inheriting contract must define its implementation.
+
+```tact
+trait FilterTrait with Ownable {
+    // Virtual functions can be overridden by users of this trait
+    virtual fun filterMessage(): Bool {
+        return sender() != self.owner;
+    }
+
+    abstract fun specialFilter(): Bool;
+
+    // Receivers can be inherited, but cannot be overridden
+    receive() { cashback(sender()) }
+}
+
+contract Filter() with FilterTrait {
+    // Overriding the default behavior of the FilterTrait
+    override fun filterMessage(): Bool {
+        return true;
+    }
+
+    override fun specialFilter(): Bool {
+        return true;
+    }
+}
+```
+
+{/* TODO: inheritance table as per https://github.com/tact-lang/tact/issues/3030 */}
+
+#### `abstract` {#attributes-abstract}
+
+<Badge text="Contract scope" title="This attribute can be applied to internal functions only" variant="note" size="medium"/><p/>
+
+The `abstract{:tact}` attribute allows declaring a [internal function](#fun-internal) with no body given at declaration. Functions with this attribute are meant to be overridden later in the inheritance hierarchy by either an intermediate trait or the contract inheriting it.
+
+```tact
+trait BlackSquare {
+    abstract fun paint();
+}
+
+trait CompositionVII {
+    abstract fun paintedWithOil(preliminary: Bool): Bool;
+}
+
+contract Abstractionism() with BlackSquare, CompositionVII {
```

Comment:
I appreciate the cultural references, but let's not do it here

we are building a DSL to handle finances, so let's keep our examples practical and definitely no string here (they are very gas expensive)

![image](https://github.com/user-attachments/assets/21e69139-ca29-469e-b5e9-a02fbf49c7d7)